yarn-project/simulator/src/avm/opcodes/context_getters.ts

```diff
@@ -0,0 +1,32 @@
+import type { AvmContext } from '../avm_context.js';
+import { Field, type MemoryValue } from '../avm_memory_types.js';
+import { Opcode } from '../serialization/instruction_serialization.js';
+import { GetterInstruction } from './instruction_impl.js';
+
+export class L2GasLeft extends GetterInstruction {
+  static type: string = 'L2GASLEFT';
+  static readonly opcode: Opcode = Opcode.L2GASLEFT;
+
+  // TODO(@spalladino) Yellow paper specifies that the value should be an Uint32, not a Field.
```

Comment:
I'd say Just send David a message on slack to confirm what the type should be.

Reply:
Thanks!